how-to/customize-workspace/client/src/actions.ts

```diff
@@ -251,5 +257,28 @@ export async function getActions(): Promise<CustomActionsMap> {
 		await authProvider.logout();
 	};
 
+	actionMap[ACTION_IDS.launchApp] = async (
+		payload: CustomButtonActionPayload | CustomDropdownItemActionPayload
+	) => {
+		if (payload.customData?.source === "dock" && payload.customData?.appId) {
```

Comment:
was thinking we keep this as a generic action (maybe someone wants to launch an app from a context menu e.g. process manager). customData.source could go into the log message if the app isn't found or data not available.

Reply:
Refactored with more error handling and more generic